libs/seiscomp/seismology/ttt/homogeneous.cpp

```diff
@@ -0,0 +1,217 @@
+ * Other Usage                                                             *
+ * Alternatively, this file may be used in accordance with the terms and   *
+ * conditions contained in a signed written agreement between you and      *
+ * gempa GmbH.                                                             *
+ ***************************************************************************/
+
+#include <seiscomp/seismology/ttt.h>
+#include <seiscomp/core/strings.h>
+#include <seiscomp/math/geo.h>
+#include <seiscomp/datamodel/config.h>
+#include <seiscomp/system/environment.h>
+
+#include <string>
+#include <cmath>
+#include <vector>
+
+using namespace std;
+using namespace Seiscomp;
+
+namespace {
+
+/**
+ * TTTHomogeneous
+ *
+ * A class to compute seismic travel times for an homogeneous (constant)
+ * velocity
+ */
+class SC_SYSTEM_CORE_API Homogeneous : public TravelTimeTableInterface {
+	public:
+		/**
+		 * Construct a TravelTimeTable object for the specified model.
+		 */
+		Homogeneous() = default;
+		~Homogeneous() = default;
+
+		Homogeneous(const Homogeneous &other) = default;
+		Homogeneous &operator=(const Homogeneous &other) = default;
+
+		Homogeneous(Homogeneous &&other) = default;
+		Homogeneous &operator=(Homogeneous &&other) = default;
+
+	public:
+		bool setModel(const std::string &model) override;
+		const std::string &model() const override;
+
+		TravelTimeList *
+		compute(double lat1, double lon1, double dep1,
+		        double lat2, double lon2, double alt2 = 0.,
+		        int ellc = 1) override;
+
+		TravelTime
+		compute(const char *phase,
+		        double lat1, double lon1, double dep1,
+		        double lat2, double lon2, double alt2 = 0.,
+		        int ellc = 1) override;
+
+		TravelTime
+		computeFirst(double lat1, double lon1, double dep1,
+		             double lat2, double lon2, double alt2 = 0.,
+		             int ellc = 1) override;
+
+		bool isInside(double lat, double lon, double dep);
+
+	private:
+		std::string _model;
+		double _pVel = 1; // [km/s]
+		double _sVel = 1; // [km/s]
+		double _centerLat = 0; // [degree]
+		double _centerLon = 0; // [degree]
+		double _radius = 0; // [km]
+		double _minDepth = 0; // [km]
+		double _maxDepth = 0; // [km]
+};
+
+double radToDeg(double r) { return 180.0 * r / M_PI; }
+
+double degToRad(double d) { return M_PI * d / 180.0; }
+
+double computeDistance(double lat1, double lon1,
+                       double lat2, double lon2,
+                       double *azimuth = nullptr,
+                       double *backAzimuth = nullptr) {
+	double dist, az, baz;
+	Math::Geo::delazi(lat1, lon1, lat2, lon2, &dist, &az, &baz);
+	dist = Math::Geo::deg2km(dist);
+	if ( azimuth ) {
+		*azimuth = az;
+	}
+	if ( backAzimuth ) {
+		*backAzimuth = baz;
+	}
+	return dist;
+}
+
+bool Homogeneous::setModel(const string &model) {
+
+	// load global configuration
+	Config::Config cfg;
+	if ( ! Environment::Instance()->initConfig(&cfg, "") ) {
```

Comment:
@gempa-jabe Could you please double check this part of the code? It works but I have the feeling there are better ways to load the global configuration.

Reply:
If an application instance is available, you can use

```c++
SCCoreApp->configGetDouble(base + "P-velocity");
```

If not or if you also want that code to be used without an application instance then this is the right way to go. It is probably not the most performant way to do it as the configuration is static and just needs to be read the first time.